File: quark/test/emit/parameterized-defaults/js/parameterized-defaults.js
var _qrt = require("datawire-quark-core");
var pkg = require('./pkg');
exports.pkg = pkg;


function main() {
    var box = new pkg.StringBox("asdf");
    _qrt.print((box).contents);
    var foo = new pkg.StringFoo();
    _qrt.print((foo).foo());
}
exports.main = main;

main();

File: quarkc/test/emit/expected/rb/parameterized-defaults/lib/parameterized_defaults.rb
module Quark
require "quark"
def self.parameterized_defaults; ParameterizedDefaults; end
module ParameterizedDefaults
require_relative 'pkg' # 0 () ()
require_relative 'parameterized_defaults' # 0 () ()

def self.call_main()
    self.main(::DatawireQuarkCore::List.new(ARGV))

    nil
end
def self.main(args)
    
    box = ::Quark.pkg.StringBox.new("asdf")
    ::DatawireQuarkCore.print((box).contents)
    foo = ::Quark.pkg.StringFoo.new()
    ::DatawireQuarkCore.print(foo.foo())


    nil
end

if __FILE__ == $0 then ::Quark.parameterized_defaults.call_main() end

end # module ParameterizedDefaults
end # module Quark
